lib/libzfs/libzfs_import.c

```diff
@@ -884,12 +912,24 @@ zpool_read_label(int fd, nvlist_t **config)
 			continue;
 		}
 
-		free(label);
-		return (0);
+		if (expected_guid) {
+			if (expected_guid == guid)
+				count++;
+
+			nvlist_free(*config);
+		} else {
+			expected_config = *config;
+			expected_guid = guid;
+			count++;
+		}
 	}
 
+	if (label != NULL)
```

Comment:
I think you meant `(labels != NULL)` here for use as a guard when assigning the count to `labels` in the subsequent line.  The memory already malloc()d for `label` must be non-NULL here.